File: templates/api-node/src/lib/cookies.ts
import type { FastifyReply } from "fastify";
import { ACCESS_COOKIE, REFRESH_COOKIE } from "./tokens.js";

export function setAuthCookies(
  reply: FastifyReply,
  tokens: { accessToken: string; refreshToken: string },
  options: { secure: boolean },
) {
  const common = {
    path: "/",
    httpOnly: true,
    secure: options.secure,
    sameSite: "lax" as const,
  };

  reply.setCookie(ACCESS_COOKIE, tokens.accessToken, {
    ...common,
    maxAge: 60 * 15,
  });
  reply.setCookie(REFRESH_COOKIE, tokens.refreshToken, {
    ...common,
    sameSite: "strict",
    maxAge: 60 * 60 * 24 * 30,
  });
}

export function clearAuthCookies(reply: FastifyReply, options: { secure: boolean }) {
  const common = {
    path: "/",
    httpOnly: true,
    secure: options.secure,
  };
  reply.clearCookie(ACCESS_COOKIE, common);
  reply.clearCookie(REFRESH_COOKIE, { ...common, sameSite: "strict" });
}

File: templates/api-python/app/lib/cookies.py
from __future__ import annotations

from fastapi import Response

from app.lib.tokens import ACCESS_COOKIE, REFRESH_COOKIE


def set_auth_cookies(
    response: Response,
    *,
    access_token: str,
    refresh_token: str,
    secure: bool,
) -> None:
    response.set_cookie(
        key=ACCESS_COOKIE,
        value=access_token,
        path="/",
        httponly=True,
        secure=secure,
        samesite="lax",
        max_age=60 * 15,
    )
    response.set_cookie(
        key=REFRESH_COOKIE,
        value=refresh_token,
        path="/",
        httponly=True,
        secure=secure,
        samesite="strict",
        max_age=60 * 60 * 24 * 30,
    )


def clear_auth_cookies(response: Response, *, secure: bool) -> None:
    response.delete_cookie(
        key=ACCESS_COOKIE,
        path="/",
        httponly=True,
        secure=secure,
        samesite="lax",
    )
    response.delete_cookie(
        key=REFRESH_COOKIE,
        path="/",
        httponly=True,
        secure=secure,
        samesite="strict",
    )
